lib/rubocop/cop/mixin/safe_mode.rb

```diff
@@ -1,22 +0,0 @@
-# frozen_string_literal: true
-
-module RuboCop
-  module Cop
-    # Common functionality for Rails safe mode.
-    module SafeMode
-      private
-
-      def rails_safe_mode?
-        safe_mode? || rails?
-      end
-
-      def safe_mode?
-        cop_config['SafeMode']
```

Comment:
RuboCop Performance 1.4 or lower users will suddenly encounter errors.

```console

#!/bin/bash -eo pipefail
bundle exec rake internal_investigation
Running RuboCop...
uninitialized constant RuboCop::Cop::Performance::Count::SafeMode
/usr/local/bundle/gems/rubocop-performance-1.4.1/lib/rubocop/cop/performance/count.rb:41:in `<class:Count>'
/usr/local/bundle/gems/rubocop-performance-1.4.1/lib/rubocop/cop/performance/count.rb:40:in `<module:Performance>'
/usr/local/bundle/gems/rubocop-performance-1.4.1/lib/rubocop/cop/performance/count.rb:5:in `<module:Cop>'
/usr/local/bundle/gems/rubocop-performance-1.4.1/lib/rubocop/cop/performance/count.rb:4:in `<module:RuboCop>'
/usr/local/bundle/gems/rubocop-performance-1.4.1/lib/rubocop/cop/performance/count.rb:3:in `<top (required)>'
/usr/local/bundle/gems/rubocop-performance-1.4.1/lib/rubocop/cop/performance_cops.rb:7:in `require_relative'
/usr/local/bundle/gems/rubocop-performance-1.4.1/lib/rubocop/cop/performance_cops.rb:7:in `<top (required)>'
/usr/local/bundle/gems/rubocop-performance-1.4.1/lib/rubocop-performance.rb:11:in `require_relative'
/usr/local/bundle/gems/rubocop-performance-1.4.1/lib/rubocop-performance.rb:11:in `<top (required)>'
```

https://circleci.com/gh/rubocop-hq/rubocop/63589

I think it is user-friendly to display a warning first. The following is an example.

```diff
 def safe_mode?
+  warn '`SafeMode` option will be removed. ' \
+       'Please update rubocop-performance 1.15.0 or higher.'
+
   cop_config['SafeMode']
 end
```

Warnings are noisy, but it's better to show the solution than sudden errors.

The next step is to remove it.

Reply:
That sounds like a good plan to me. Do we want to specify the version of RuboCop that we will be removing the `safe_mode?` method from? That way it gives people a time frame of when they need to make the change as well.

Reply:
Good! The following is an example of milestones:

- RuboCop 0.75 (next release) : Warn `safe_mode?`
- RuboCop 0.76 : Remove `safe_mode?`

Reply:
Is there a better place for this deprecation warning to live? Adding the message here will add the warning to the terminal every time that `safe_mode?` is invoked. Since this happens inside of `on_send` in both `Performance/Detect` and `Performance/Count`, the output of RuboCop becomes unreadable rather quickly. 

I've pushed an update that includes a class variable to prevent the message from being printed multiple times. It seems like there should be a better solution to this problem.

Maybe this check could be moved to where we load the plugins into RuboCop.